boto/dynamodb2/table.py

```diff
@@ -407,39 +412,100 @@ def update(self, throughput, global_indexes=None):
+            ...             HashKey('requiredHashkey', data_type=STRING),
+            ...             RangeKey('optionalRangeKey', data_type=STRING)
+            ...         ],
+            ...         throughput={
+            ...             'read': 2,
+            ...             'write': 1,
+            ...         }),
+            ...      }
+            ... })
+
+            # To delete a global index(es)
+            >>> users.update(global_secondary_indexes={
+            ...     'delete': 'TheIndexNameHere'
             ... })
             True
 
         """
-        self.throughput = throughput
-        data = {
-            'ReadCapacityUnits': int(self.throughput['read']),
-            'WriteCapacityUnits': int(self.throughput['write']),
-        }
+
+        data = None
+
+        if throughput:
+            self.throughput = throughput
+            data = {
+                'ReadCapacityUnits': int(self.throughput['read']),
+                'WriteCapacityUnits': int(self.throughput['write']),
+            }
+
         gsi_data = None
+        gsi_data_attr_def = None
 
         if global_indexes:
             gsi_data = []
 
-            for gsi_name, gsi_throughput in global_indexes.items():
-                gsi_data.append({
-                    "Update": {
-                        "IndexName": gsi_name,
-                        "ProvisionedThroughput": {
-                            "ReadCapacityUnits": int(gsi_throughput['read']),
-                            "WriteCapacityUnits": int(gsi_throughput['write']),
+            for gsi_operation_key, gsi_operation_value in global_indexes.items():
+                if gsi_operation_key == 'delete':
+
+                    gsi_data.append({
+                        "Delete": {
+                            "IndexName": gsi_operation_value
+                        },
+                    })
+
+                elif gsi_operation_key == 'update':
+
+                    for gsi_update_key, gsi_throughput in gsi_operation_value.items():
+                        gsi_data.append({
+                            "Update": {
+                                "IndexName": gsi_update_key,
+                                "ProvisionedThroughput": {
+                                    "ReadCapacityUnits": int(gsi_throughput['read']),
+                                    "WriteCapacityUnits": int(gsi_throughput['write']),
+                                },
+                            },
+                        })
+                elif gsi_operation_key == 'create':
+
+                    gsi_data.append({
+                        "Create": gsi_operation_value.schema()
+                    })
+
+                    gsi_data_attr_def = []
+
+                    for attr_def in gsi_operation_value.parts:
+                        gsi_data_attr_def.append(attr_def.definition())
+                else:
+                    # Provide backward compatibility for the old dict structure
```

Comment:
You wouldn't have to provide backward compatibility anymore. But: it would be good to keep the comment you wrote, for reference.